npep/npep-122.md

```diff
@@ -0,0 +1,194 @@
+    
+    Example: I would like to define two tenants in my cluster, one composed of the pods in foo-ns-1 and foo-ns-2 
+    and the other with pods in bar-ns-1, where inter-tenant traffic is denied.
+
+From the User Stories:
+
+    Story 4: Create and Isolate multiple tenants in a cluster
+    
+    As a cluster admin, I want to build tenants in my cluster that are isolated from each other by default. 
+    Tenancy may be modeled as 1:1, where 1 tenant is mapped to a single Namespace, or 1:n, where a single tenant 
+    may own more than 1 Namespace.
+
+Elsewhere:
+
+    AdminNetworkPolicy Pass rules allows an admin to delegate security posture for certain traffic to the Namespace 
+    owners by overriding any lower precedence Allow or Deny rules. For example, intra-tenant traffic management can be 
+    delegated to tenant admins explicitly with the use of Pass rules.
+    
+So really, the only solidly-agreed-upon use case is that you should be able to create enforced isolation 
+between particular sets of namespaces.
+
+### Clarifying user stories
+
+#### Story 4.1: Create and Isolate multiple tenants in a cluster by default
+
+Here is the existing tenancy related user story from our website:
+
+    As a cluster admin, I want to build tenants in my cluster that are isolated from each other by default. 
+    Tenancy may be modeled as 1:1, where 1 tenant is mapped to a single Namespace, or 1:n, where a single tenant 
+    may own more than 1 Namespace.
+
+The wording used here has lead to much confusion. Specifically, "By default" part actually means this policy should 
+be at the BANP priority, and solves the "Engineering org vs Marketing org" use case where you want to keep orgs 
+from accidentally interfering with each other, but still want to allow them to communicate if needed. 
+
+In reality, the CR attached to this story defines strict tenancy, which should be a separate use case (see next section).
+
+To make the use case more obvious we can add some details to it like the following
+
+    As a cluster admin, I want to build tenants in my cluster that are isolated from each other by default.
+    This should prevent tenants from interfering with each other by accident, but may be overridden by namespace
+    owners when needed.
+    Tenancy may be modeled as 1:1, where 1 tenant is mapped to a single Namespace, or 1:n, where a single tenant 
+    may own more than 1 Namespace.
+
+#### Story 4.2: Create and Isolate multiple tenants in a cluster, strict isolation
+
+Strict tenancy is the "Coke vs Pepsi" sort of thing where you want each tenant to feel like it has its own cluster, 
+and be totally independent of the other tenants. We can write it down like this
+
+    As a cluster admin, I want to build tenants in my cluster that are isolated from each other and this can't
+    be overridden. This policy should make every tenant completely independent and isolated from another tenants.
```

Comment:
```suggestion
    As a cluster admin, I want to build tenants in my cluster that are isolated from each other where this
    isolation can't be overridden by the tenants themselves. This policy should make every tenant completely independent and isolated from another tenants.
```

(because the admin could still override the isolation with ANPs)

alternatively "where the isolation can't be overridden by NetworkPolicy". Actually, that might be a good way to describe both the previous case and this one; the previous case is where the tenant isolation _can_ be overridden by NetworkPolicy and this case is where it _can't_ be.

Reply:
good point! I made it "namespace owners" just based on the first user story wording, lmk if you prefer some other name for that actor :)